docs/contributor/testing-strategy.md

```diff
@@ -0,0 +1,16 @@
+# Testing Strategy
+
+Each pull request to the repository triggers the following CI/CD jobs:
+
+- `golangci-lint / lint (pull_request)` - Is responsible for linting and static code analysis.
+- `PR Markdown Link Check / markdown-link-check (pull_request)` - Checks if there are no broken links in the pull request `.md` files.
+- `Run unit tests / validate (pull_request)` - Executes basic create/update/delete functional tests of the reconciliation logic.
+- `Run vuln check / test (pull_request)` - Runs [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) on the code to detect known vulnerabilities.
+- `pull-infrastructure-manager-build` - builds the Docker image and pushes it to the registry.
+
+After the pull request is merged, the following CI/CD jobs are executed:
+
+ - `Run unit tests / validate (push)` - Executes basic create/update/delete functional tests of the reconciliation logic.
+ - `Run vuln check / test (push)` - Runs [govulncheck](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) on the code to detect known vulnerabilities.
+ - `golangci-lint / lint (push)` - Is responsible for linting and static code analysis.
+ - `main-infrastructure-manager-build` - Rebuilds the image and pushes it to the registry.
```

Comment:
I'm a prow novice and I don't exactly know what is the difference between `pull-infrastructure-manager-build` and `main-infrastructure-manager-build`. Anyway I've added link to the responsible configuration file.
```suggestion
 - `main-infrastructure-manager-build` - Rebuilds the image and pushes it to the registry. It's configured [here](https://github.com/kyma-project/test-infra/blob/a3c2a07da4ba42e468f69cf42f1960d7bfcc3fff/prow/jobs/kyma-project/infrastructure-manager/infrastructure-manager.yaml).
```

Reply:
Maybe @ameteiko will be able to show the differences :)